File: frontend/src/services/analytics.js
import { getToken, getBusinessId } from './auth';

function headers() {
  const h = { 'Content-Type': 'application/json' };
  const t = getToken();
  if (t) h['Authorization'] = `Bearer ${t}`;
  const b = getBusinessId();
  if (b) h['X-Business-Id'] = b;
  return h;
}

async function req(path) {
  const res = await fetch(path, { headers: headers() });
  if (res.status === 401) { window.location.href = '/login'; throw new Error('Session expired'); }
  const ctype = res.headers.get('content-type') || '';
  if (!res.ok) {
    const txt = await res.text();
    if (ctype.includes('text/html')) throw new Error(`Backend not reachable on ${path}`);
    let msg = txt;
    try { msg = JSON.parse(txt).detail || txt; } catch {}
    throw new Error(msg);
  }
  return res.json();
}

export const pipelineVelocity = () => req('/api/analytics/pipeline-velocity');
export const revenueForecast = (months = 6) => req(`/api/analytics/revenue-forecast?horizon_months=${months}`);
export const agentImpact = (days = 30) => req(`/api/analytics/agent-impact?days=${days}`);
export const churnRisk = (maxDeals = 15) => req(`/api/analytics/churn-risk?max_deals=${maxDeals}`);

File: api/routers/analytics.py
"""
Analytics endpoints — pipeline velocity, revenue forecast, agent impact,
churn risk. All read-only aggregates over the tenant's CRM + invoice data.
"""
from __future__ import annotations

from fastapi import APIRouter, Depends

from api.auth import get_current_context
from api import analytics as _analytics

router = APIRouter(tags=["analytics"])


@router.get("/api/analytics/pipeline-velocity")
def analytics_pipeline_velocity(ctx: dict = Depends(get_current_context)):
    return _analytics.pipeline_velocity(ctx["business_id"])


@router.get("/api/analytics/revenue-forecast")
def analytics_revenue_forecast(horizon_months: int = 6,
                               ctx: dict = Depends(get_current_context)):
    horizon_months = max(1, min(12, horizon_months))
    return _analytics.revenue_forecast(ctx["business_id"],
                                       horizon_months=horizon_months)


@router.get("/api/analytics/agent-impact")
def analytics_agent_impact(days: int = 30,
                           ctx: dict = Depends(get_current_context)):
    days = max(1, min(180, days))
    return _analytics.agent_impact(ctx["business_id"], days=days)


@router.get("/api/analytics/churn-risk")
def analytics_churn_risk(max_deals: int = 15,
                         ctx: dict = Depends(get_current_context)):
    max_deals = max(1, min(50, max_deals))
    return _analytics.churn_risk(ctx["business_id"], max_deals=max_deals)
